### DISCUSSION TITLE
What is the equivalent of Latex's \pageref?

### DISCUSSION THREAD

**Discussion by Victor-N-Suadicani** [typst/typst#2142] [Q&A] [answered]
I'm struggling to see how I can write "See the glossary on page \<page number>". Given a label, I want to find the page the label is on, to reference the label by page number instead of section numbering. How do I do this?

**Comment by EpicEricEE** [comment, marked as answer]
You could do something like
```typ
#let pageref(label) = locate(loc => {
  let label-loc = query(label, loc).first().location()
  let page-numbering = label-loc.page-numbering()
  if page-numbering == none { page-numbering = "1" }
  link(label-loc, numbering(page-numbering, label-loc.page()))
})

See the glossary on page #pageref(<glossary>)

= Glossary <glossary>
...
```

**Reply by laurmaedje** [reply]
You can also directly use `label-loc` for the link. It will be a bit more efficient.
